src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -20221,9 +20223,9 @@ void Unit::CastDelayedSpellWithPeriodicAmount(Unit* caster, uint32 spellId, Aura
 
     // xinef: delay only for casting on different unit
     if (this == caster || !sWorld->getBoolConfig(CONFIG_MUNCHING_BLIZZLIKE))
-        caster->CastCustomSpell(spellId, SPELLVALUE_BASE_POINT0, addAmount, this, TriggerCastFlags(TRIGGERED_FULL_MASK & ~TRIGGERED_NO_PERIODIC_RESET), nullptr, nullptr, caster->GetGUID());
+        caster->CastCustomSpell(spellId, SPELLVALUE_BASE_POINT0, addAmount, this, TriggerCastFlags(TRIGGERED_FULL_MASK & ~TRIGGERED_NO_PERIODIC_RESET), nullptr, aurEff, caster->GetGUID());
     else
-        caster->m_Events.AddEvent(new AuraMunchingQueue(*caster, GetGUID(), addAmount, spellId), caster->m_Events.CalculateQueueTime(400));
+        caster->m_Events.AddEvent(new AuraMunchingQueue(*caster, GetGUID(), addAmount, spellId, aurEff), caster->m_Events.CalculateQueueTime(400));
```

Comment:
@heyitsbench Here is your crash related issue.
As you can see, you are adding an event for the future (+400ms, if I'm not mistaken) and capturing the `auraEff` pointer. 
But what would happen if, for example, after 200ms, the aura is removed and the auraEff pointer is deleted? Then, in 400ms the `AuraMunchingQueue` tries to use `auraEff`, but you SHOULD NOT use it, since `auraEff` pointer points to a deleted object in the memory.

One way to solve this is in the AuraMunchingQueue::Execute function, where you can validate whether the auraEff pointer is still valid. You can do this by iterating over the `m_modAuras` of the `_owner `unit and checking if the captured `auraEff` pointer is still in the list. If the `auraEff` pointer is not in the list, it means that the object was deleted and removed from the list.

Reply:
https://github.com/azerothcore/azerothcore-wotlk/compare/master...heyitsbench:munching-redux
``
Here's my attempt at implementing what you're suggesting here. Gave a quick test and the behavior I aimed to fix originally is still being corrected, but I was never able to reproduce the crash locally, so I can't confirm whether it fixes that yet. Anything you can point out is immediately wrong/poorly done?